be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/DbActor.scala

```diff
@@ -137,157 +180,269 @@ object DbActor extends AskPatternConstants {
+          //val objectType = message.decodedData.getOrElse(null)._object //or some other way, then match using match case and establish channel type
+          val objectType = ObjectType.LAO
+          val actualObjectType = if (objectType == ObjectType.ELECTION || objectType == ObjectType.CHIRP){
+            objectType
+          } else {
+            ObjectType.LAO
+          }
+          createChannel(channel, actualObjectType) match {
+            case DbActorAck() =>
+            case _ =>
+              log.info(s"Actor $self (db) encountered a problem while creating channel '$channel'")
+              DbActorNAck(ErrorCodes.SERVER_ERROR.id, s"Error when creating channel '$channel'")
+          }
       }
-
+      
       val messageId: Hash = message.message_id
-      Try(channelDb.put(messageId.getBytes, message.toJsonString.getBytes)) match {
-        case Success(_) =>
-          log.info(s"Actor $self (db) wrote message_id '$messageId' on channel '$channel'")
-          DbActorWriteAck
+      Try(db.get(channel.toString.getBytes)) match {
+        case Success(bytes) if bytes != null => {
+          val key: String = channel + "/" + messageId.toString
+          Try(db.createWriteBatch()) match {
+            case Success(batch) => {
+              val json = new String(bytes, StandardCharsets.UTF_8)
+              batch.put(channel.toString.getBytes, ChannelData.buildFromJson(json).addMessage(messageId).toJsonString.getBytes)
+              batch.put(key.getBytes, message.toJsonString.getBytes)
+              //allows writing all data atomically
+              Try(db.write(batch)) match {
+                case Success(_) =>
+                  log.info(s"Actor $self (db) wrote object 'objectId' and message_id '$messageId' on channel '$channel'") //change with object and objectId/name/smth like that
+                  DbActorWriteAck()
+                case Failure(exception) =>
+                  log.info(s"Actor $self (db) encountered a problem while writing message_id '$messageId' and object 'objectId' on channel '$channel' because of the batch write")
+                  DbActorNAck(ErrorCodes.SERVER_ERROR.id, exception.getMessage)
+              }      
+            }
+            case Failure(exception) =>
+              log.info(s"Actor $self (db) encountered a problem while writing message_id '$messageId' and object 'objectId' on channel '$channel' because of a batch creation")
+              DbActorNAck(ErrorCodes.SERVER_ERROR.id, exception.getMessage)
+          }
+        }
         case Failure(exception) =>
           log.info(s"Actor $self (db) encountered a problem while writing message_id '$messageId' on channel '$channel'")
           DbActorNAck(ErrorCodes.SERVER_ERROR.id, exception.getMessage)
+        case _ =>
+          log.info(s"Actor $self (db) encountered a problem while writing message_id '$messageId' on channel '$channel', as the channel does not exist")
```

Comment:
I'd recommend turning this log into `log.error`

Reply:
good catch